File: ModernizacionPersonas.Web/ClientApp/src/app/models/slip.ts
import { EdadAsegurabilidad } from './edad-asegurabilidad';
import { ActionResponseBase } from './action-response.base';
import { Vigencia } from './fichatecnica';
import { CotizacionState } from 'src/app/models';

export interface Slip {
  codigoCotizacion: number;
  estadoCotizacion: CotizacionState;
  codigoRamo: number;
  codigoSubramo: number;
  fechaString: string;
  infoEncabezado: SlipPageHeader;
  imagenProductoUri: string;
  imagenFirmaUri: string;
  descripcion: string;
  ciudad: string;
  fecha: string;
  tomador: SlipTomador;
  tomadorIntermediario: SlipTomador;
  asunto: string;
  tipoPoliza: string;
  diasVigencia: number;
  diasVigenciaWords: string;
  vigencia: Vigencia;
  dirigidoSlip: DirigidoSlip;
  infoGeneral: SlipInformacionGeneralSection;
  amparos: SlipAmparosSection;
  gruposAsegurados: SlipGruposAsegurados;
  clausulas: SlipClausulasSection;
  condiciones: SlipCondicionesSection;
  disposiciones: SlipDisposicionesSection;
  informacionEnvio: string[];
  nombreSucursal: string;
}

export interface SlipGruposAsegurados {
  gruposAsegurados: SlipGrupoAsegurado[];
  tituloSeccion: string;
}

export interface SlipGrupoAsegurado {
  codigoGrupoAsegurado: number;
  nombre: string;
  snTasaMensual: boolean;
  edades: EdadesAmparoSlip[];
  valorMaximo: ValorMaximoAseguradoIndividualSlip;
  valoresAmparos: ValoresAseguradosAmparoSlip[];
}

export interface SlipSeccion {
  codigoSeccion: string;
  codigoAmparo: string;
  tituloSeccion: string;
  texto: string;
}

export interface SlipPageHeader {
  headerImageUri: string;
  footerImageUri: string;
  imagen1Uri: string;
  imagen2Uri: string;
}

export interface SlipTomador {
  nombre: string;
  codigoTipoDocumento: number;
  numeroDocumento: string;
  actividad: string;
  direccion: string;
  telefono: string;
  departamento: string;
  ciudad: string;
}

export interface EdadesAmparoSlip {
  nombreAmparo: string;
  edadMinimaIngreso: number;
  edadMaximaIngreso: number;
  edadMaximaPermanencia: number;
}

export interface ValorMaximoAseguradoIndividualSlip {
  secciones: string;
  tituloSeccion: string;
}

export interface ValoresAseguradosAmparoSlip {
  codigoTipoSumaAsegurada: number;
  nombreAmparo: string;
  countOpciones: number;
  opciones: OpcionValoresAseguradosAmparoSlip[];
}

export interface OpcionValoresAseguradosAmparoSlip {
  valorAseguradoIndividual?: number;
  valorAsegurado: number;
  valorTasaComercial: number;
  tipoValor: number;
  tasaMensual: number;
  numeroDias: number;
  tablaValoresDiarios: boolean;
  valorDiario: boolean;
}

export interface SlipAmparosSection {
  tituloSeccion: string;
  amparos: SlipSeccion[];
}

export interface SlipClausulasSection {
  tituloSeccion: string;
  secciones: SlipSeccion[];
  asegurabilidad: EdadAsegurabilidad[];
}

export interface SlipCondicionesSection {
  tituloSeccion: string;
  secciones: string;
}

export interface SlipInformacionGeneralSection {
  tituloSeccion: string;
  secciones: SlipSeccion[];
}

export interface SlipDisposicionesSection {
  tituloSeccion: string;
  secciones: SlipSeccion[];
}

export interface GenerarSlipDataResponse extends ActionResponseBase {
  data: Slip;
}

export interface DirigidoSlip {
  nombre: string;
  direccion: string;
  telefono: string;
}

File: ModernizacionPersonas.Entities/Slip.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace ModernizacionPersonas.Entities
{
    public class Slip
    {
        public int CodigoRamo { get; set; }

        public int CodigoSubramo { get; set; }
        public PageHeaderInfo InfoEncabezado { get; set; }
        public string ImagenProductoUri { get; set; }
        public string Descripcion { get; set; }
        public string Ciudad { get; set; }
        public DateTime Fecha { get; set; }
        public TomadorSlip Tomador { get; set; }
        public TomadorSlip TomadorIntermediario { get; set; }
        public DirigidoSlip DirigidoSlip { get; set; }
        public string Asunto { get; set; }
        public string TipoPoliza { get; set; }
        public int DiasVigencia { get; set; }
        public string DiasVigenciaWords { get; set; }
        public string NombreSucursal { get; set; }
        public VigenciaSlip Vigencia { get; set; }
        public InfoGeneralTextosSeccionSlip InfoGeneral { get; set; }
        public AmparosTextosSeccionSlip Amparos { get; set; }
        public GruposAseguradosSlip GruposAsegurados { get; set; }
        public EdadesSlipSection Edades { get; set; }
        public ClausulasTextosSeccionSlip Clausulas { get; set; }
        public CondicionesTextosSeccionSlip Condiciones {get; set;}
        public DisposicionesTextosSeccionSlip Disposiciones { get; set; }
        public IEnumerable<string> InformacionEnvio { get; set; }
    }

    public class VigenciaSlip
    {
        public DateTime? Desde { get; set; }

        public DateTime? Hasta { get; set; }
    }

    public class PageHeaderInfo
    {
        public string HeaderImageUri { get; set; }
        public string FooterImageUri { get; set; }
        public string Imagen1Uri { get; set; }
        public string Imagen2Uri { get; set; }
    }
}
